### PR TITLE
Installation Script and Documentation Fixes

### PR DESCRIPTION
**Pull request by danintel**
Installation Script Fixes

These installation fixes were made after a developer reviewed and tested the
installation documentation.

`rebuild.sh`
* Rename SRCDIR to TCF_HOME
* Don't set `proxy type` and `aesm proxy` in `aesmd.conf` if there is no proxy

Signed-off-by: danintel <daniel.anderson@intel.com>

Documentation Fixes

These installation fixes were made after a developer reviewed and tested the
installation documentation.

`PREREQUISITES.md`
* Add `cd /var/tmp` before wget to avoid permission errors
* Must wget libsgx-enclave-common
* Update proxy instructions now that proxy is not set by default

`tcs_config.toml`
* Replace Intel proxy by placeholder: `http://your-proxy:your-port/`
* Replace ias_api_key with a dummy key
* Wrap comment lines at 80 characters and reword.

Signed-off-by: danintel <daniel.anderson@intel.com>

### CODE REVIEW

File: PREREQUISITES.md [hunk 1]
```diff
@@ -226,7 +233,7 @@ ias_api_key = '<ias subscription key obtained from portal>'
 ```
 
 In the same file, if you are behind a corporate proxy,
-update the https_proxy line:
+uncomment and update the https_proxy line:
 
 ```
 https_proxy = "http://your-proxy:your-port/"
```
[the remark this answers is not in the document]
Comment (danintel) [reply PREREQUISITES.md]: Done.

File: tools/rebuild.sh [hunk 2]
```diff
@@ -62,23 +65,26 @@ function try() {
...
 if [[ ${SGX_MODE} &&  "${SGX_MODE}" == "HW" ]]; then
-    # Add proxy settings
-    echo "proxy type = manual" >> /etc/aesmd.conf
-    echo "aesm proxy = $http_proxy" >> /etc/aesmd.conf
+
+    # Add proxy settings, if proxy is present and not set
+    grep -qs "^aesm proxy" /etc/aesmd.conf
+    if [ $? -ne 0 -a -n "$http_proxy" ] ; then
+       echo "Setting AESMD proxy"
+       echo "proxy type = manual" >> /etc/aesmd.conf
```
[the remark this answers is not in the document]
Comment (danintel) [reply tools/rebuild.sh]: Done.

File: config/tcs_config.toml [hunk 3]
```diff
@@ -49,7 +49,7 @@ num_of_enclaves = "1"
 ias_url = "https://api.trustedservices.intel.com/sgx/dev"
 
 # Proxy for https. Change to your corporate proxy or comment out below line for direct Internet connections.
-https_proxy = "http://proxy-iind.intel.com:912/"
+#https_proxy = "http://your-proxy:your-port/"
 
 # IAS api key is subscription key used for authentication of requests submitted to IAS server.
 # this can be obtained by subscribing in the portal https://api.portal.trustedservices.intel.com.
```
Comment (arsulegai) [inline config/tcs_config.toml]: Is this a valid ias_api_key? Can we replace it with the dummy value here?

File: PREREQUISITES.md [hunk 4]
```diff
@@ -174,12 +174,18 @@ install the Intel SGX driver for both standalone and docker builds.
 You need to install the Intel SGX driver whether you build TCF standalone
```
Comment (arsulegai) [inline PREREQUISITES.md]: Could you please segregate these changes into 2 commits?

File: config/tcs_config.toml (hunk 3 above)
Comment (danintel) [reply config/tcs_config.toml]: I do not know if the example key is valid.  I replaced it with a dummy value.

File: PREREQUISITES.md (hunk 4 above)
Comment (danintel) [reply PREREQUISITES.md]: I separated the commits into 2 sets:
-  Installation documentation changes in `PREREQUISITES.md` and `tcs_config.toml`
- Installation changes in `rebuild.sh`

**Pull request merged**